src/metax_api/services/common_service.py

```diff
@@ -281,3 +284,71 @@ def _get_object_for_update(model_obj, row, results, check_modified_since):
                 pass
 
         return instance
+
+    @staticmethod
+    def validate_and_get_if_unmodified_since_header_as_tz_naive_datetime(request):
+        try:
+            return CommonService._validate_http_date_header(request, 'HTTP_IF_UNMODIFIED_SINCE')
+        except:
+            raise Http400('Bad If-Unmodified-Since header')
+
+    @staticmethod
+    def validate_and_get_if_modified_since_header_as_tz_naive_datetime(request):
+        try:
+            return CommonService._validate_http_date_header(request, 'HTTP_IF_MODIFIED_SINCE')
+        except:
+            raise Http400('Bad If-Modified-Since header')
+
+    @staticmethod
+    def _validate_http_date_header(request, header_name):
+        timestamp = request.META.get(header_name, '')
+        # According to RFC 7232, Http date should always be expressed in 'GMT'. Forcing its use makes this explicit
+        if not timestamp.endswith('GMT'):
+            raise Exception
+        return parse_http_timestamp_using_tz_and_get_tz_naive_datetime(timestamp)
+
+    @staticmethod
+    def get_request_header(request, header_name):
+        return request.META.get(header_name, None)
+
+    @staticmethod
+    def request_has_header(request, header_name):
+        return header_name in request.META
+
+    @staticmethod
+    def request_is_write_operation(request):
+        return request.method in ('POST', 'PUT', 'PATCH', 'DELETE')
+
+    @staticmethod
+    def check_if_unmodified_since(request, obj):
+        if CommonService.request_is_write_operation(request) and \
+                CommonService.request_has_header(request, 'HTTP_IF_UNMODIFIED_SINCE'):
+
+            header_timestamp = CommonService.validate_and_get_if_unmodified_since_header_as_tz_naive_datetime(request)
+            CommonService._check_if_obj_unmodified_since(obj, header_timestamp)
+
+    @staticmethod
+    def _check_if_obj_unmodified_since(obj, header_timestamp):
```

Comment:
Questioning the necessity for this method. The code "if obj.modified_since(header_timestamp)": and what follows should be well understandable without the information conveyed by the name of this method. Basically looks like a "method to call a method"